From 5d0fcfb0cb58407ed493770bdbc22245b90ca468 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Sat, 22 Apr 2023 22:39:05 +0300
Subject: [PATCH] GLODROID: sun8i_codec: Change headset button detection
 threshold

Change BTN1 threshold to v<5, so the BTN2 can be triggered on v=5

Change-Id: I6af6760896c7701a68853d8b2fe0bd515a2a56bd
Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 sound/soc/sunxi/sun8i-codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/sunxi/sun8i-codec.c b/sound/soc/sunxi/sun8i-codec.c
index 342b097b5249..bed40864601e 100644
--- a/sound/soc/sunxi/sun8i-codec.c
+++ b/sound/soc/sunxi/sun8i-codec.c
@@ -1665,7 +1665,7 @@ static irqreturn_t sun8i_codec_jack_irq(int irq, void *dev_id)
 
 			if (value < 0x2)
 				type |= SND_JACK_BTN_0;
-			else if (value < 0x7)
+			else if (value < 0x5)
 				type |= SND_JACK_BTN_1;
 			else if (value < 0x10)
 				type |= SND_JACK_BTN_2;
-- 
2.39.2

